### DISCUSSION TITLE
Download workflow run logs using Github API returns invalid zip file

### DISCUSSION THREAD

**Discussion by kerbou** [community/community#52602] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Hi

I am experiencing unexpected behaviour when downloading a workflow log using Github CLI. 

**Example URL:**

gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/{myrepo}/actions/runs/{runId}/logs >c:\temp\log.zip

**Expected behaviour**
I can open the file using 7-zip or similar tool

**Actual behaviour**
Windows tells me the that the compressed file is invalid

**Additional info**
Opening the file, I can see that it contains binary content - by the looks of it, it looks like a binary zipfile, since I can see job and step names within the binary content.

I have tried multiple API endpoints for downloading content, all files downloaded seems to be invalid,

Can somebody please look into this?

**Comment by frottke** [comment]
I have the same issue, did you find a solution for this?

**Comment by paulie-of-punskas** [comment]
Hey. I didn't encounter it on Windows 11, however I used [curl](https://curl.se) for that: 

```bash
curl --location \
--header "Accept: application/vnd.github+json" \ 
--header "Authorization: Bearer <your_token>" \
--header "X-GitHub-Api-Version: 2022-11-28" \ 
https://api.github.com/repos/{organization}/{repo}/actions/runs/{run_id}/logs \
--output gha_log.zip
```

I successfully could open downloaded file with Total Commander and with Windows Explorer.

I got similar error to what @kerbou described, when I downloaded it on Red Hat 9.2, with Python `requests.get()`. Errors I got, were:

```
caution: zipfile comment truncated
error: missing <string> bytes in zipfile (attempting to process anyway)
error: missing start of central directory not found; zipfile corrupt
```

I fixed it, by using `'wb'` instead of `'a'`, when writing to a disk:

```python
import requests

your_token = ''
organization = ''
repo = ''
run_id = ''

headers = {
    'Accept': 'application/vnd.github+json',
    'Authorization': f'Bearer {your_token}',
    'X-GitHub-Api-Version': '2022-11-28'
}

url = f"https://api.github.com/repos/{organization}/{repo}/actions/runs/{run_id}/logs"
response = requests.get(url=url, headers=headers)

with open('./gha_log.zip', 'wb') as file:
    file.write(response.content)
``` 

I used `unzip gha_log.zip` command to unpack it.
